git-hooks/commit-msg.py

```diff
@@ -49,23 +45,30 @@ def __init__(self, msg, *, strict=False):
             (self.Rules.RULE12, self.validate_rule12),
         ]
 
-        self.strict = strict
         self.enough = False
 
-        # Strip comments and trailing whitespace
-        self.msg = re.sub(r"^#.*\n?", "", msg, flags=re.MULTILINE)
-        self.msg = self.msg.rstrip()
+        self.msg = subprocess.check_output(
+            ["git", "stripspace", "-s"], input=msg, text=True
+        )
 
         self.lines = self.msg.splitlines() if msg else []
         self.subject = self.lines[0].rstrip() if self.lines else ""
         self.body = self.lines[1:]
 
-        if not self.strict:
-            # In non-strict mode, allow special keywords used by `git rebase --autosquash`
-            self.subject = re.sub(r"^(fixup|squash)! ", "", self.subject)
-
         self.errors = self.validate_all()
 
+    def should_ignore_line(self, line):
```

Comment:
Initially I was going to propose moving this inside `validate_rule12` or renaming to `should_ignore_length`. That would make the current scope clearer. But then I figured that we may need this in other rules as well, for the same reasons, and would naturally reuse the function. So how about renaming it as something like:

- `is_external_line`
- `is_constrained_line`
- `is_unavoidable_line`
- `is_modifiable_line`
- `is_uncontrollable_line`
- `is_untouchable_line`
- `is_inflexible_line` 

It took me a little to land on it, but I think `is_inflexible_line` is the closest to what we want to express. It leaves it open to reuse, without implying that it should be ignored in general. WDYT?

Reply:
`inflexible` sounds good to me, too. I'll update the function name.